src/components/PostDetails.tsx

```diff
@@ -1,92 +1,60 @@
-import React, { useEffect, useState } from 'react';
+import React, { useEffect, useRef, useState } from 'react';
 import { Loader } from './Loader';
+import { useAppSelector, useAppDispatch } from '../app/hooks';
 import { NewCommentForm } from './NewCommentForm';
-
-import * as commentsApi from '../api/comments';
-
-import { Post } from '../types/Post';
-import { Comment, CommentData } from '../types/Comment';
+import { CommentData } from '../types/Comment';
+import {
+  fetchComments,
+  removeComment,
+  addComment,
+} from '../features/comments/commentsSlice';
+import { useSelector } from 'react-redux';
 
 type Props = {
   post: Post;
 };
 
+interface Post {
+  id: number;
+  title: string;
+  body: string;
+}
+
 export const PostDetails: React.FC<Props> = ({ post }) => {
-  const [comments, setComments] = useState<Comment[]>([]);
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
+  const dispatch = useAppDispatch();
+  const {
+    loaded,
+    hasError,
+    items: comments,
+  } = useAppSelector(state => state.comments);
   const [visible, setVisible] = useState(false);
+  const previousPostId = useRef(post.id);
```

Comment:
The `previousPostId` ref seems redundant since the `useEffect` already resets visibility when the post ID changes. Consider removing it to simplify the code.